src/Plugin/Block/BlogPrevNextBlock.php

```diff
@@ -0,0 +1,243 @@
+      return Cache::mergeTags(parent::getCacheTags(), ['node:' . $node->id()]);
+    }
+    else {
+      // Return default tags instead.
+      return parent::getCacheTags();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['route']);
+  }
+
+  /**
+   * Lookup the previous node,youngest node which is still older than the node.
+   *
+   * @param string $node
+   *   Show current page node id.
+   *
+   * @return array
+   *   A render array for a previous node.
+   */
+  private function generatePrevious($node) {
+    return $this->generateNextPrevious($node, 'prev');
+  }
+
+  /**
+   * Lookup the next node,oldest node which is still younger than the node.
+   *
+   * @param string $node
+   *   Show current page node id.
+   *
+   * @return array
+   *   A render array for a next node.
+   */
+  private function generateNext($node) {
+    return $this->generateNextPrevious($node, 'next');
+  }
+
+  const DIRECTION__NEXT = 'next';
+
+  /**
+   * Lookup the next or previous node.
+   *
+   * @param string $node
+   *   Get current page node id.
+   * @param string $direction
+   *   Default value is "next" and other value come from
+   *   generatePrevious() and generatePrevious().
+   *
+   * @return array
+   *   Find the alias of the next node.
+   */
+  private function generateNextPrevious($node, $direction = self::DIRECTION__NEXT) {
+    $comparison_opperator = '>=';
+    $sort = 'ASC';
+    $current_node_id = $node->id();
+    $current_node_date = $node->get('localgov_blog_date')->value;
+    $current_langcode = $node->get('langcode')->value;
+    $current_blog_channel = $node->get('localgov_blog_channel')->target_id;
+
+    if ($direction === 'prev') {
+      $comparison_opperator = '<=';
+      $sort = 'DESC';
+    }
+
+    // Lookup 1 node younger (or older) than the current node
+    // based upon the `localgov_blog_date` field.
+    $storage = $this->entityTypeManager->getStorage('node');
+    $query_result = $storage->getQuery();
+    $results = $query_result->condition('localgov_blog_date', $current_node_date, $comparison_opperator)
+      ->condition('type', 'localgov_blog_post')
+      ->condition('localgov_blog_channel', $current_blog_channel)
+      ->condition('status', 1)
+      ->condition('langcode', $current_langcode)
+      ->sort('localgov_blog_date', $sort)
+      ->sort('created', $sort)
+      ->sort('nid', $sort)
+      ->accessCheck(TRUE)
+      ->execute();
+
+    // Since sometimes the next / prev blog post could be on the same day, and
+    // the query will not have sufficent granularity yet, we need to retrive
+    // an array of all following / previous including the current post.
+    // We can then search in the array and find the next one in the list.
+    // @todo possibly remove if localgov_blog_date changed to a datetime field.
+    $results = array_values($results);
```

Comment:
So with @andybroomfield change we've got all the results, and  then work with the ordered array. But we loose the range(0, 1) so this could have a lot of results.

Probably needs a check. But at least we don't loose blog posts that are on the same day here.